### PR TITLE
feat(buck2): complete TypeScript authority

### PR DESCRIPTION
**Pull request by schickling**
## What

- admit all 39 TypeScript projects to package-local Buck authority, including the React Inspector strict consumer and five bootstrap-critical projects
- make Buck the sole declaration producer for 35 package surfaces and atomically publish their dist trees
- delete root tsconfig check/emit graphs and source TypeScript tasks from devenv, CI, and editor commands
- move Buck runtime ownership into @overeng/buck2-tools and remove obsolete compatibility paths

## Evidence

- complete synthetic authority build: 39 projects, 72 local commands, 47.33s, BUILD SUCCEEDED
- all 35 declaration publishers completed end to end in 89.65s
- strict consumer target passed independently
- hostile socket mutation failed with TS2322; restored target passed
- authority/runner suites: 67/67; publisher rollback cases: 4/4; package command: 2/2; baseline collection: 51/51
- direct shipped-equivalent oxlint: 0 warnings, 0 errors
- genie --check and staged whitespace check passed

## Review

Three independent review/deep-clean passes found and fixed the schema-form-aria output path, missing required test input, hidden runner diagnostics, projection-wrapper coverage, namespace collisions, duplicate authority handling, dead compatibility branches, and stale dashboard/editor surfaces.

## Constraints

Local Nix execution remains deferred because / is below the host policy's 200 GB free-space floor after the authorized GC service ran. CI must exercise the Nix-backed materializer guard tests. This PR remains draft and must be reviewed and merged manually.

### CODE REVIEW

**Comment by schickling-assistant** [comment]
@codex Please review this pull request as part of Buck2 stack #1270. Focus on correctness, migration authority boundaries, generated-file ownership, and safe downstream behavior. Review against the cumulative stack below this pull request.


<details>
<summary>Posted on behalf of @schickling</summary>

| field | value |
| --- | --- |
| `agent_identity` | dev3.direct.omp.pnakg447 |
| `session` | unknown |
| `agent_persona` | generalist |
| `agent_supervisor` | unavailable |
| `agent_tool` | OMP |
| `agent_tool_version` | 18.1.14 |
| `agent_runtime` | OMP 18.1.14 |
| `tooling_profile` | dotfiles@5d1e05f |
</details>

**Comment by schickling-assistant** [comment]
@codex review


<details>
<summary>Posted on behalf of @schickling</summary>

| field | value |
| --- | --- |
| `agent_identity` | dev3.direct.omp.pnakg447 |
| `session` | unknown |
| `agent_persona` | generalist |
| `agent_supervisor` | unavailable |
| `agent_tool` | OMP |
| `agent_tool_version` | 18.1.14 |
| `agent_runtime` | OMP 18.1.14 |
| `tooling_profile` | dotfiles@5d1e05f |
</details>

File: AGENTS.md [hunk 1]
```diff
@@ -6,19 +6,19 @@
 
 Use `devenv tasks run <task>` (devenv tasks) to execute tasks with dependencies:
 
-- **TypeScript**: `devenv tasks run ts:check` or `devenv tasks run ts:build-watch` (watch mode) or `devenv tasks run ts:clean`
+- **TypeScript**: `devenv tasks run buck2:check`
+- **Declaration publication**: `devenv tasks run buck2:typescript:materialize-dist`
```
[the remark this answers is not in the document]
Comment (schickling-assistant) [reply AGENTS.md:10]: Fixed in a963df154: the terminal stack commit refreshes the affected documentation to the current TypeScript authority and task names. The fix is included in PR #1283.

File: .github/workflows/ci.yml.genie.ts [hunk 2]
```diff
@@ -429,15 +429,11 @@ const nativeDepPolicyAuditStep = {
...
-  //   - `ts:check:strict` owns only the residual root TypeScript solution (the projects with
-  //     no Buck authority). It consumes Buck-owned declarations through the dist overlays that
-  //     `buck2:typescript:materialize-dist` publishes, so it is ordered after the materializer.
+  // Buck is the single TypeScript check and declaration authority.
   typecheck: job({
     step: {
-      name: 'Type check (Buck authority + residual root solution)',
-      run: runDevenvTasksBefore('buck2:check', 'ts:check:strict'),
+      name: 'Type check (Buck)',
+      run: runDevenvTasksBefore('buck2:check'),
```
[the remark this answers is not in the document]
Comment (schickling-assistant) [reply .github/workflows/ci.yml.genie.ts:446]: Fixed on cumulative head bb10b1fe8. gh-ci-utils is now in buck2TypeScriptAdmissions and has generated Buck typecheck, dist, test, editor, and product targets.
